lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -983,36 +1019,36 @@ class MongoClientInterface {
     deleteOrRepairPHD(c, bucketName, objName, mst, vFormat, log, cb) {
         const masterKey = formatMasterKey(objName, vFormat);
         // Check if there are other versions available
-        this.getLatestVersion(c, objName, vFormat, log, err => {
-            if (err) {
-                if (err === errors.NoSuchKey) {
-                    // We try to delete the master. A race condition
-                    // is possible here: another process may recreate
-                    // a master or re-delete it in between so place an
-                    // atomic condition on the PHD flag and the mst
-                    // version:
-                    // eslint-disable-next-line
-                    c.findOneAndDelete({
-                        '_id': masterKey,
-                        'value.isPHD': true,
-                        'value.versionId': mst.versionId,
-                    }, {}, err => {
-                        if (err) {
-                            log.error(
-                                'findOneAndDelete: error finding and deleting',
-                                { error: err.message });
-                            return cb(errors.InternalError);
-                        }
-                        // do not test result.ok === 1 because
-                        // both cases are expected
-                        return cb(null);
-                    });
-                    return undefined;
-                }
+        this.getLatestVersion(c, objName, vFormat, log, (err, version) => {
+            if (err && err !== errors.NoSuchKey) {
                 log.error('getLatestVersion: error getting latest version',
                     { error: err.message });
                 return cb(err);
             }
+            if (err === errors.NoSuchKey || version.isDeleteMarker) {
```

Comment:
Same comment about the use of PHD in v1: the metadata implementation does not create PHDs anymore in v1, so make sure that this change is because Mongo still creates PHDs in v1 and that it's wanted this way.